source/server/hot_restart_nop_impl.h

```diff
@@ -14,7 +17,18 @@ namespace Server {
  */
 class HotRestartNopImpl : public Server::HotRestart {
 public:
-  HotRestartNopImpl() {}
+  HotRestartNopImpl() {
+    int num_stats = 100;
```

Comment:
@mrice32 TODO(jbuckland) find a less arbitrary number for this. I seem to recall setting this to `ENVOY_DEFAULT_MAX_STATS` caused testing to hang. What should we do here?

Reply:
This is not a test, this is the real application, so this *must* be pulled from the supplied command line options/default just like the hot restart implementation does.  I would suggest supplying it via the constructor just like HotRestartImpl: 
```
HotRestartImpl(Options& options);
```